File: node/routes/question.js
import express from "express";
import { createClient } from "@supabase/supabase-js";
import { embed } from "../util/embed.js";
const route = express.Router();
const supabase = createClient(process.env.SUPABASE_URL, process.env.SUPABASE_KEY);

route.post("/", async (req, res) => {
    try {
        const { repoId, question } = req.body;
        if (!repoId || !question) {
            return res.status(400).json({ error: "Repository ID and question are required" })
        }
        let { data, error } = await supabase.from("repos").select("status").eq("id", repoId);
        
        
        if (data[0].status != "success") {
            return res.status(400).json({ error: "Repository is not ready yet" })
        }
        if (error) {
            return res.status(500).json({ error: "Internal server error" })
        }
        const embedding = await embed(question);
        if (embedding.error) {
            return res.status(500).json({ error: "Internal server error" })
        }
        let { data: top_chunks, error: e } = await supabase.rpc("match_embeddings", {
            query_embedding: embedding.embeddings[0],
            match_count: 10,
            repo_id_input: repoId
        });

        if (e) {
            console.error(e);
        }

        res.json({ top_chunks })

    } catch (error) {
        console.log(error);
        res.status(500).json({ error: "Internal server error" })
    }
})
export default route;

File: python/routes/question.py
from fastapi import APIRouter
from supabase import create_client
import os
from dotenv import load_dotenv
from services.question_chain import process_question

load_dotenv()

supabase = create_client(os.getenv("SUPABASE_URL"), os.getenv("SUPABASE_KEY"))
router = APIRouter()

@router.post("/")
def question(body: dict):
    repo_id = body["repo_id"]
    question = body["question"]
    if not repo_id or not question:
        return {"error": "Missing repo_id or question"}

    response = supabase.table("repos").select("status").eq("id", repo_id).execute()

    if not response.data or response.data[0]["status"] != "success":
        return {"error": "Repo not processed yet or processing has failed"}
    
    response = process_question(repo_id, question)
    return {"response": response}